### PR TITLE
First batch of edits to middleware docs

### PR DESCRIPTION
**Pull request by StephenBarlow**
Not done but think these edits are all ok standalone

### CODE REVIEW

File: docs/source/integrations/middleware.md [hunk 1]
```diff
@@ -1,47 +1,158 @@
...
+</tbody>
+</table>
+
+## Swapping out `apollo-server`
+
+After you get up and running with the "batteries-included" `apollo-server` package, you might want to configure its HTTP behavior in ways that this package doesn't support. For example, you might want to run some middleware before processing GraphQL requests, or you might want to serve other endpoints from the same server.
+
+In this case, we recommend you swap out `apollo-server` for `apollo-server-express` (unless you are confident that you want to use a different Node.js framework). This change requires only a few lines and has a minimal effect on your server's existing behavior (`apollo-server` uses `apollo-server-express` under the hood).
+
+> We recommend Express because it's the most popular Node.js web framework, and it integrates well with many _other_ popular frameworks. It does have its limitations (for example, Express async support is not built around `Promise`s and `async` functions), but backward incompatible changes to the framework are rarer than in newer frameworks.
```
Comment (glasser) [inline docs/source/integrations/middleware.md:120]: s/frameworks/libraries/

File: docs/source/integrations/middleware.md [hunk 2]
```diff
@@ -1,47 +1,158 @@
...
+</td>
+</tr>
+</tbody>
+</table>
+
+## Swapping out `apollo-server`
+
+After you get up and running with the "batteries-included" `apollo-server` package, you might want to configure its HTTP behavior in ways that this package doesn't support. For example, you might want to run some middleware before processing GraphQL requests, or you might want to serve other endpoints from the same server.
+
+In this case, we recommend you swap out `apollo-server` for `apollo-server-express` (unless you are confident that you want to use a different Node.js framework). This change requires only a few lines and has a minimal effect on your server's existing behavior (`apollo-server` uses `apollo-server-express` under the hood).
```
Comment (glasser) [inline docs/source/integrations/middleware.md:118]: Do we want to have a section here that directly shows how to swap out (ie what lines of code to write)? @trevor-scheer was saying that it felt like that was lacking and this seems like the right spot. (Not a regression in this sub-PR, shouldn't block merge)

File: docs/source/integrations/middleware.md [hunk 3]
```diff
@@ -51,16 +162,20 @@ async function startApolloServer(typeDefs, resolvers) {
...
+`apollo-server-express` is the Apollo Server package for [Express](https://expressjs.com/), the most popular Node.js web framework. It enables you to attach a GraphQL server to an existing Express server.
+
+`apollo-server` uses `apollo-server-express` under the hood. If you start with `apollo-server` and later need to modify how it serves over HTTP, you can [swap `apollo-server` to `apollo-server-express`](#swapping-out-apollo-server).
 
 The following function is roughly equivalent to the `apollo-server` sample code above.
 
-```
-$ npm install apollo-server-express express graphql
+```bash
+npm install apollo-server-express express graphql
```
Comment (glasser) [inline docs/source/integrations/middleware.md:178]: I can't put the comment directly on it but yikes, the line `const { url } = await server.listen();` should NOT be in the apollo-server-express example

**Review by glasser — APPROVED** [review]
otherwise looks great!

**Comment by glasser** [comment]
Something else we need to include (in this PR or on mine) is that `apollo-server` hosts the operation on `/` but all the others host on `/graphql` by default.  (Which kinda sucks and I probably should have fixed, to make the GraphQL path just be part of how you attach it to your server rather than built in to the middleware. But I didn't.)

And specifically, the "swap" instructions should have you call `server.applyMiddleware({ app, path: '/' })` to preserve the `apollo-server` behavior

File: docs/source/integrations/middleware.md (hunk 2 above)
Comment (StephenBarlow) [reply docs/source/integrations/middleware.md:118]: Definitely! I'm going to leave out of this PR and add to next

**Pull request merged**